python/social_bot/envs/pr2.py

```diff
@@ -58,128 +81,288 @@ def __init__(self,
-        joint_state = gazebo.JointState(1)
-        joint_state.set_positions([1.4])
-        joint_state.set_velocities([0.0])
-        self._collision_cnt = 0
-        self._agent.set_joint_state("pr2::pr2::head_tilt_joint", joint_state)
-        self._world.step(200)
+        self._move_goal()
+        self._grip = False
 
+        if self._adjust_position_at_start:
+            # move camera to focus on ground. it is hacky
+            joint_state = gazebo.JointState(1)
+            joint_state.set_positions([1.4])
+            joint_state.set_velocities([0.0])
+            self._agent.set_joint_state("pr2::pr2::head_tilt_joint", joint_state)
+
+            self._agent.take_action(dict({"pr2::pr2::r_gripper_joint":5}))
+
+        self._world.step(200)
         self._goal = self._world.get_agent(self._goal_name)
-        self._move_goal()
+
+        obs = self._get_observation()
+
+        self._table_height = self._goal_pose[0][2]
+        #logger.debug("table height:" + str(self._table_height))
         self._steps_in_this_episode = 0
         self._cum_reward = 0.0
+
         self._prev_dist = self._get_finger_tip_distance()
-        return self._get_observation()
+        self._prev_gripper_pos = self._get_gripper_pos()
+        self._gripper_reward_dir = 1
+        return obs
 
     def _move_goal(self):
-        loc = (0.6 + 0.3 * (random.random() - 1), 0.4 * (random.random() - 1),
-               0)
+        loc = (0.76 + 0.14 * (random.random() - 1),  0.1 * (random.random() - 1.0),
+               0.43)
         self._goal.set_pose((loc, (0, 0, 0)))
 
     def _get_observation(self):
-        obs = np.array(
-            self._agent.get_camera_observation(
-                "default::pr2::pr2::head_tilt_link::head_mount_prosilica_link_sensor"
-            ),
-            copy=False)
-        obs = PIL.Image.fromarray(rgb2gray(obs)).resize(
-            self._resized_image_size, PIL.Image.ANTIALIAS)
-
-        obs = np.reshape(np.array(obs),
-                         [self._resized_image_size[0], self._resized_image_size[1], 1])
+        self._l_touch = self._finger_tip_contact("r_gripper_l_finger_tip_contact_sensor",
+                                           "beer::link::box_collision")
+        self._r_touch = self._finger_tip_contact("r_gripper_r_finger_tip_contact_sensor",
+                                           "beer::link::box_collision")
+        self._goal_pose = self._goal.get_pose()
+        self._l_finger_pose = self._agent.get_link_pose("pr2::pr2::r_gripper_l_finger_tip_link")
+        self._r_finger_pose = self._agent.get_link_pose("pr2::pr2::r_gripper_r_finger_tip_link")
+
+        if self._use_internal_states_only:
+            joint_states = [self._agent.get_joint_state(joint) for joint in self._r_arm_joints]
+            joint_positions = [s.get_positions()[0] for s in joint_states]
+            joint_velocities = [s.get_velocities()[0] for s in joint_states]
+            loc, loc_a = self._goal_pose
```

Comment:
should consider using logical camera to get the relative pose of the objects in the view